File: ReloadedAndMobilized/ReloadedAndMobilizedTests/ServiceRequestTests.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace SquishySquidward.ReloadedAndMobilized
{

    public class ServiceRequestTests
    {
        [TestFixture]
        public class NewRequest
        {
            private static readonly ServiceKind kind = ServiceKind.Janitor;
            private readonly ServiceRequest request = new ServiceRequest(kind);

            [Test]
            public void HasNoAssignedEmployee()
            {
                Assert.That(request.Employee, Is.Null);
            }

            [Test]
            public void HasStatusCreated()
            {
                Assert.That(request.Status, Is.EqualTo(ServiceStatus.Created));
            }

            [Test]
            public void HasSpecifiedServiceKind()
            {
                Assert.That(request.Kind, Is.EqualTo(ServiceKind.Janitor));
            }

            [Test]
            public void ThrowsWhenAssignedToNullEmployee()
            {
                Assert.Throws<ArgumentNullException>(() => request.Assign(null));
            }
        }

        [TestFixture]
        public class AssignedRequest
        {
            private ServiceRequest request;
            private Employee bob;

            [SetUp]
            public void SetUp()
            {
                request = new ServiceRequest(ServiceKind.Janitor);
                bob = new Employee { FirstName = "Bob" };
                request.Assign(bob);
            }

            [Test]
            public void HasAssignedEmployee()
            {
                Assert.That(request.Employee, Is.EqualTo(bob));
            }

            [Test]
            public void HasStatusAssigned()
            {
                Assert.That(request.Status, Is.EqualTo(ServiceStatus.Assigned));
            }
        }

        [TestFixture]
        public class CompletedRequest
        {
            private ServiceRequest request;
            private Employee alice;

            [SetUp]
            public void SetUp()
            {
                request = new ServiceRequest(ServiceKind.Janitor);
                alice = new Employee { FirstName = "Alice" };
                request.Assign(alice);
                request.Complete();
            }

            [Test]
            public void HasStatusCompleted()
            {
                Assert.That(request.Status, Is.EqualTo(ServiceStatus.Completed));
            }

            [Test]
            public void HasAssignedEmployee()
            {
                Assert.That(request.Employee, Is.EqualTo(alice));
            }
        }

        [TestFixture]
        public class CanceledRequest
        {
            private ServiceRequest request;
            
            [SetUp]
            public void SetUp()
            {
                request = new ServiceRequest(ServiceKind.Janitor);
                request.Assign(new Employee());
                request.Cancel();
            }

            [Test]
            public void HasStatusCanceled()
            {
                Assert.That(request.Status, Is.EqualTo(ServiceStatus.Canceled));
            }

            [Test]
            public void HasNullAssignedEmployee()
            {
                Assert.That(request.Employee, Is.Null);
            }
        }
    }
}

File: src/test/java/edu/wpi/cs3733/c20/teamS/serviceRequests/ServiceRequestTests.java
package edu.wpi.cs3733.c20.teamS.serviceRequests;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class ServiceRequestTests {
    private static class MockRequest extends ServiceRequest {
        public MockRequest() {
            super();
        }
        public void accept(ServiceVisitor visitor) {
            throw new UnsupportedOperationException();
        }
    }
    public static class NewRequest {
        private final ServiceRequest request = new MockRequest();

        @Test
        public void hasStatus_created() {
            assertEquals(ServiceStatus.CREATED, request.status());
        }

        @Test
        public void hasAssignee_null() {
            assertNull(request.assignee());
        }
    }

    public static class AssignedRequest {
        private final Employee employee;
        private final ServiceRequest request;

        public AssignedRequest() {
            employee = new Employee(420, "Alice");
            request = new MockRequest();
            request.assignTo(employee);
        }

        @Test
        public void hasStatus_Assigned() {
            assertEquals(ServiceStatus.ASSIGNED, request.status());
        }

        @Test
        public void hasAssignedEmployee() {
            assertEquals(employee, request.assignee());
        }
    }

    public static class CompletedRequest {
        private final Employee employee;
        private final ServiceRequest request;

        public CompletedRequest() {
            employee = new Employee(69, "Bob");
            request = new MockRequest();
            request.assignTo(employee);
            request.complete();
        }

        @Test
        public void hasStatus_Completed() {
            assertEquals(ServiceStatus.COMPLETED, request.status());
        }

        @Test
        public void hasAssignedEmployeeAtCompletionTime() {
            assertEquals(employee, request.assignee());
        }
    }

    public static class CanceledRequest {
        private final ServiceRequest request;

        public CanceledRequest() {
            request = new MockRequest();
            request.assignTo(new Employee(42, "Amy"));
            request.cancel();
        }

        @Test
        public void hasStatus_Canceled() {
            assertEquals(ServiceStatus.CANCELED, request.status());
        }

        @Test
        public void hasNullAssignee() {
            assertNull(request.assignee());
        }
    }
}
